# PT002

`configuration for fixture '{name}' specified via positional args, use kwargs`

## Examples

Bad code:

```python
import pytest

@pytest.fixture('module')
def my_fixture():
    ...
```

Good code:

```python
import pytest

@pytest.fixture(scope='module')
def my_fixture():
    ...
```

## Rationale

* to make parameters meaning more obvious
* to enforce consistency between all fixtures in a codebase
