# PT009

`use a regular assert instead of unittest-style '{assertion}'`

## Examples

Bad code:

```python
import unittest

class TestFoo(unittest.TestCase):
    def test_foo(self):
        self.assertEqual(a, b)
```

Good code:

```python
import unittest

class TestFoo(unittest.TestCase):
    def test_foo(self):
        assert a == b
```

## Rationale

* to enforce the assertion style recommended by pytest
* to make use of pytest's assertion rewriting
