# PT003

`scope='function' is implied in @pytest.fixture()`

Fixtures with function scope should not specify scope explicitly because
function scope is implied by default.

## Examples

Bad code:

```python
import pytest

@pytest.fixture(scope='function')
def my_fixture():
    ...
```

Good code:

```python
import pytest

@pytest.fixture()
def my_fixture():
    ...
```

## Rationale

* to enforce consistency between all fixtures in a codebase
