# PT010

`set the expected exception in pytest.raises()`

## Examples

Bad code:

```python
import pytest

def test_foo():
    with pytest.raises():
        do_something()
```

Good code:

```python
import pytest

def test_foo():
    with pytest.raises(SomeException):
        do_something()
```


## Rationale

* not passing the exception class will fail at runtime
