# PT019

`fixture {name} without value is injected as parameter, use @pytest.mark.usefixtures instead`

## Examples

Bad code:

```python
def test_foo(_patch_something):
    ...
```

Good code:
```python
import pytest

@pytest.mark.usefixtures('_patch_something')
def test_foo():
    ...
```

## Rationale

* to avoid unused parameters in test functions
