# PT013

`found incorrect import of pytest, use simple 'import pytest' instead`

## Examples

Bad code:

```python
import pytest as pt
from pytest import fixture
```

Good code:

```python
import pytest
```

## Rationale

* to enforce consistency
* to help the linter check usage of members imported from `pytest`
