# PT021

`use yield instead of request.addfinalizer`

`pytest` offers two ways to perform cleanup in fixture code.  One is sequential
using the `yield` statement, and the other is callback-based.

## Examples

Bad code:

```python
import pytest

@pytest.fixture()
def my_fixture(request):
    resource = acquire_resource()
    request.addfinalizer(resource.release)
    return resource
```

Good code:
```python
import pytest

@pytest.fixture()
def my_fixture():
    resource = acquire_resource()
    yield resource
    resource.release()
```

## Rationale

* to make fixture code more linear and straightforward
