# PT020

`@pytest.yield_fixture is deprecated, use @pytest.fixture`

## Examples

Bad code:

```python
import pytest

@pytest.yield_fixture()
def my_fixture():
    obj = SomeClass()
    yield obj
    obj.cleanup()
```

Good code:

```python
import pytest

@pytest.fixture()
def my_fixture():
    obj = SomeClass()
    yield obj
    obj.cleanup()
```

## Rationale

* to avoid using the deprecated function
